### PR TITLE
Pedro/bal 384 import pool modal

### PR DESCRIPTION
**Pull request by yvesfracari**
This PR:
- Refactor GQL queries (fewer queries and more generics)
- Refactor Search Pool Form (make it easy to use in other contexts)
- Add the import pool modal with the function to load the pool variables
- Impacts the metadata application as well

https://www.loom.com/share/41d16418dc8c4bd7adead14ebc6d3d04

### CODE REVIEW

File: packages/gql/src/balancer-pools/pools.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { gql } from "graphql-tag";
 
 export const poolWhereOwner = gql`
-  query Pools($owner: Bytes!) {
+  query PoolsWhereOwner($owner: Bytes!) {
```
Comment (ribeirojose) [inline packages/gql/src/balancer-pools/pools.ts:4]: `#would` call this only `Pools`

File: packages/gql/src/balancer-pools/pools.ts [hunk 2]
```diff
@@ -15,26 +15,43 @@ export const poolWhereOwner = gql`
...
+    $poolId: ID!
+    $poolTypes: [String!] = [
+      "Weighted"
+      "ComposableStable"
+      "Stable"
+      "MetaStable"
+      "Element"
+      "LiquidityBootstrapping"
+      "Linear"
+    ]
```
Comment (ribeirojose) [inline packages/gql/src/balancer-pools/pools.ts:29]: 👀 `TIL`

File: packages/next-app/src/app/metadata/layout.tsx [hunk 3]
```diff
@@ -33,7 +36,12 @@ export default function Layout({ children }: React.PropsWithChildren) {
         <div className="flex flex-1 gap-x-8">
           <div>
             <Sidebar isFloating>
-              <Dialog title="Go  to pool" content={<SearchPoolForm />}>
+              <Dialog
+                title="Go to pool"
+                content={
+                  <SearchPoolForm onSubmit={usePoolMetadata().handleGoToPool} />
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/layout.tsx:42]: 👀

File: packages/next-app/src/contexts/PoolMetadataContext.tsx [hunk 4]
```diff
@@ -88,6 +93,16 @@ export function PoolMetadataProvider({ children }: PropsWithChildren) {
...
+  function handleGoToPool(formData: PoolAttribute) {
+    const { push } = useRouter();
+    const { chain } = useNetwork();
+
+    const networkId = formData.network ?? chain?.id.toString();
+    const networkName =
+      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
+    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+  }
+
```
Comment (ribeirojose) [inline packages/next-app/src/contexts/PoolMetadataContext.tsx:105]: Not sure I follow why define this function within the context -- are we using anything that's context dependent here?

File: packages/next-app/src/components/WalletNotConnected.tsx [hunk 5]
```diff
@@ -19,7 +20,12 @@ export default function WalletNotConnected({
         <div className="text-center text-slate11 text-lg">
           {/* TODO: */}
           Alternatively, open a pool directly clicking&nbsp;
-          <Dialog title="Go to pool" content={<SearchPoolForm />}>
+          <Dialog
+            title="Go to pool"
+            content={
+              <SearchPoolForm onSubmit={usePoolMetadata().handleGoToPool} />
```
Comment (ribeirojose) [inline packages/next-app/src/components/WalletNotConnected.tsx:26]: 👀

File: packages/next-app/src/components/SearchPoolForm.tsx [hunk 6]
```diff
@@ -33,33 +38,27 @@ export default function SearchPoolForm({ close }: { close?: () => void }) {
...
 
-  function handleSubmitForm(formData: PoolMetadataAttribute) {
-    const networkId = formData.network ?? chain?.id.toString();
-    const networkName =
-      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
-    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+  const isPool = !!poolData?.pools?.length;
 
+  function handleSubmitForm(formData: PoolAttribute) {
+    onSubmit?.(formData);
```
Comment (ribeirojose) [inline packages/next-app/src/components/SearchPoolForm.tsx:56]: 👍 Great addition -- I'd add that if `onSubmit` is not defined this should probably throw an error

File: packages/next-app/src/components/SearchPoolForm.tsx [hunk 7]
```diff
@@ -33,33 +38,27 @@ export default function SearchPoolForm({ close }: { close?: () => void }) {
...
-      poolId,
-    },
-    { revalidateIfStale: true }
-  );
-
-  const isPool = poolAddress?.pool?.address ? true : false;
+  const gqlVariables = poolTypeFilter?.length
+    ? { poolId, poolTypes: poolTypeFilter }
+    : { poolId };
+  const { data: poolData } = pools
```
Comment (ribeirojose) [inline packages/next-app/src/components/SearchPoolForm.tsx:49]: this is probably `poolsData` right?

File: packages/next-app/src/components/SearchPoolForm.tsx [hunk 8]
```diff
@@ -33,33 +38,27 @@ export default function SearchPoolForm({ close }: { close?: () => void }) {
...
+    .gql(network || "1")
+    .usePoolsWherePoolTypeInAndId(gqlVariables, { revalidateIfStale: true });
```
Comment (ribeirojose) [inline packages/next-app/src/components/SearchPoolForm.tsx:51]: Do we really need to revalidate? 🤔 when would this get stale?

File: packages/next-app/src/components/SearchPoolForm.tsx [hunk 9]
```diff
@@ -33,33 +38,27 @@ export default function SearchPoolForm({ close }: { close?: () => void }) {
...
-    {
-      poolId,
-    },
-    { revalidateIfStale: true }
-  );
-
-  const isPool = poolAddress?.pool?.address ? true : false;
+  const gqlVariables = poolTypeFilter?.length
+    ? { poolId, poolTypes: poolTypeFilter }
+    : { poolId };
```
Comment (ribeirojose) [inline packages/next-app/src/components/SearchPoolForm.tsx:48]: 👍 Could also be 
```suggestion
  const gqlVariables = { poolId, ...(poolTypeFilter.length ? {poolTypes: poolTypeFilter} : {}) }
```

File: packages/gql/src/balancer-pools/pools.ts (hunk 1 above)
Comment (yvesfracari) [reply packages/gql/src/balancer-pools/pools.ts:4]: Merge this with which query?

File: packages/next-app/src/contexts/StableSwapContext.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+          rate: token?.priceRate,
+        })) || [],
+    };
+  }
+
+  async function handleImportPoolParametersById(formData: PoolAttribute) {
+    const poolData = await pools.gql(formData.network || "1").Pool({
+      poolId: formData.poolId,
+    });
+    setInitialData(convertGqlToAnalysisData(poolData));
```
Comment (ribeirojose) [inline packages/next-app/src/contexts/StableSwapContext.tsx:50]: to make convertGqlToAnalysisData less complicated you can also do
```suggestion
    if (!poolData) return
    setInitialData(convertGqlToAnalysisData(poolData));
```

File: packages/next-app/src/components/Header.tsx [hunk 11]
```diff
@@ -17,9 +17,10 @@ interface IHeader {
   imageSrc?: string;
   title: string;
   children?: ReactNode;
+  wallet?: boolean;
 }
 
-export function Header({ linkUrl, imageSrc, title, children }: IHeader) {
+export function Header({ linkUrl, imageSrc, title, children, wallet=true }: IHeader) {
```
Comment (ribeirojose) [inline packages/next-app/src/components/Header.tsx:23]: prettier didn't run here 😅

File: packages/next-app/src/contexts/PoolMetadataContext.tsx (hunk 4 above)
Comment (yvesfracari) [reply packages/next-app/src/contexts/PoolMetadataContext.tsx:105]: It is not, but it was a way to share the function between the layout (sidebar) and the pages. I could also put it inside a handler folder. WDYT?

**Review by ribeirojose — DISMISSED** [review]

File: packages/next-app/src/contexts/PoolMetadataContext.tsx (hunk 4 above)
Comment (ribeirojose) [reply packages/next-app/src/contexts/PoolMetadataContext.tsx:105]: I'd suggest only keep things in the context that need to be there and if want to share that in two files just separate it into another file

**Pull request merged**